src/transformers/pipelines/image_text_to_text.py

```diff
@@ -0,0 +1,165 @@
+        """
+        Generate a text given text and the image(s) passed as inputs.
+
+        Args:
+            images (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a HTTP(s) link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images.
+
+            text (`str`):
+                The text to be used as a prompt for the generation.
+
+            max_new_tokens (`int`, *optional*):
+                The amount of maximum tokens to generate. By default it will use `generate` default.
+
+            generate_kwargs (`Dict`, *optional*):
+                Pass it to send all of these arguments directly to `generate` allowing full control of this function.
+            timeout (`float`, *optional*, defaults to None):
+                The maximum time in seconds to wait for fetching images from the web. If None, no timeout is set and
+                the call may block forever.
+
+        Return:
+            A list or a list of list of `dict`: Each result comes as a dictionary with the following key:
+
+            - **generated_text** (`str`) -- The generated text.
+        """
+        return super().__call__(images, **kwargs)
+
+    def preprocess(self, image, text=None, timeout=None):
+        image = load_image(image, timeout=timeout)
+
+        model_type = self.model.config.model_type
+
+        kwargs = {}
+        if model_type == "pix2struct":
+            kwargs = {"add_special_tokens": False}
+
+        model_inputs = self.processor(images=image, text=text, return_tensors=self.framework, **kwargs)
+
+        if model_type == "git":
+            # remove EOS token from input_ids and attention_mask
+            model_inputs["input_ids"] = model_inputs["input_ids"][:, :-1]
+            model_inputs["attention_mask"] = model_inputs["attention_mask"][:, :-1]
+
+        if model_type == "vision-encoder-decoder" and self.processor.__class__.__name__ == "DonutProcessor":
+            model_inputs["decoder_input_ids"] = self.processor.tokenizer(
+                text, add_special_tokens=False, return_tensors="pt"
+            ).input_ids
+
+        return model_inputs
+
+    def _forward(self, model_inputs, generate_kwargs=None):
+        if generate_kwargs is None:
+            generate_kwargs = {}
+
+        model_outputs = self.model.generate(**model_inputs, **generate_kwargs)
+        return model_outputs
+
+    def postprocess(self, model_outputs):
+        records = []
+        for output_ids in model_outputs:
+            record = {
+                "generated_text": self.processor.decode(
+                    output_ids,
+                    skip_special_tokens=True,
+                )
+            }
+            records.append(record)
+        return records
```

Comment:
For KOSMOS-2 for instance there's an additional step for postprocessing like following:
`processed_text = processor.post_process_generation(generated_text, cleanup_and_extract=False)`
Which may be why the error occurs, I'll try to see if it's the reason why

Reply:
Ok this for sure fixes it. I'll make the necessary suggestion in the PR

Reply:
Yes I tried it with KOSMOS-2, and for now I wouldn't add this specific method to the pipeline, as only KOSMOS-2 has it. For now it's recommended for users to call this `post_process_generation` method themselves after getting the output of the pipeline.

In the future, I'm considering having a general `post_process_generation` method for multimodal models which require additional postprocessing after batch_decode. cc @ydshieh

Reply:
Is Kosmos-2 still the only model whose generation's text would containing tags (like bounding box info. as string)?

Reply:
Yes, although Donut/Nougat also have some quite specific postprocessing: https://github.com/huggingface/transformers/blob/main/docs/source/en/model_doc/donut.md#inference-examples